tests/41end-to-end-keys/01-upload-key.pl

```diff
@@ -35,6 +47,35 @@
       })
   };
 
+test "Rejects invalid device keys",
+   requires => [ $fixture ],
+
+   proves => [qw( can_upload_e2e_keys )],
+
+   do => sub {
+      my ( $user ) = @_;
+
+      do_request_json_for( $user,
+         method  => "POST",
+         uri     => "/r0/keys/upload",
+         content => {
+            device_keys => {
```

Comment:
Can you please add an inline comment here to state that the reason that this is invalid is because it is missing required fields, for the next person looking at this?